modules/op-specifying-pipelines-resource-quota-using-priority-class.adoc

```diff
@@ -167,13 +166,16 @@ Ensure that all tasks in the pipeline belongs to the same priority class.
 .Example: YAML for a pipeline run
 [source,yaml]
 ----
-apiVersion: tekton.dev/v1alpha1
+apiVersion: tekton.dev/v1beta1
 kind: PipelineRun
 metadata:
   generateName: petclinic-run-
 spec:
```

Comment:
This a doc about how to use priority class but its' not defined anywhere. I tried to specify it in pipeline run
```
spec:
  podTemplate:
    priorityClassName: pipeline1-pc
```

but the run failed with following error. 

```
failed to create task run pod "petclinic-run-r7m29-build": pods "petclinic-run-r7m29-build-pod" is forbidden: failed quota: pipeline1-rq: must specify cpu for: prepare,step-git-source-source-dqghc,step-mvn,working-dir-initializer; memory for: prepare,step-git-source-source-dqghc,step-mvn,working-dir-initializer. Maybe missing or invalid Task asdf/mvn
```

Any idea how to fix it @vdemeester?

Reply:
@ppitonak the error you get is about a quota error though 🤔

Reply:
https://tekton.dev/docs/pipelines/podtemplates/ `priorityClass` is defined there though 🤔

Reply:
@vdemeester The note with an error @ppitonak mentioned is right before the chapter we are working with: ([Alternative approaches for limiting compute resource consumption in OpenShift Pipelines](https://57466--docspreview.netlify.app/openshift-enterprise/latest/cicd/pipelines/setting-compute-resource-quota-for-openshift-pipelines.html#alternative-approaches-compute-resource-quota-pipelines_setting-compute-resource-quota-for-openshift-pipelines)). I wonder what would be the best approach here. Maybe to move the note to the procedure so that it is more visible to the user?  Or should we change the example ResourceQuota to include a limit range (if I understand correctly that the limit range is a solution here)?  WDYT?

Reply:
I am not entirely sure I follow 🤔 😅 
cc @concaf

Reply:
This document is talking about how to limit resource quota used by pipeline run.

1. the current version *does not do anything*
2. the document says that we would achieve the goal using priority class but priority class is not defined anywhere
3. when I define priority class, it fails - IMHO caused by the fact that we are missing range limit
4. I think that priority class is not the correct feature to achieve this goal

Reply:
@ppitonak i dug up my proposal doc which i wrote while adding these docs in downstream; where you ask that why is LimitRange not enough i found this:

> Why can we not use LimitRange?
A LimitRange can put constraints on pods or individual containers in a given namespace - but the scope is the entire namespace; there is no way to configure resources for a subset of pods in a namespace. So, it cannot distinguish pods running for a Pipeline from the rest of the pods.
For example, a Maven build task in a Pipeline requires much more resources than the pod for the resulting application. App operators need a way to be able to give enough quota for the Maven build for that application, without giving too much quota to all pods in that namespace.

> LimitRange applies to all pods and does not distinguish between Tekton and other pods. If a Pipeline runs resource-intensive builds, the customer has to increase the limit for all pods via the LimitRange in that namespace which is not acceptable for the customer.

Reply:
here is the doc btw - https://docs.google.com/document/d/1WTnkhI8wkn2rY7q_5q8BCGG9JD1_qPLiL2Y2ryOmHc4/edit